File: asset-management/src/main/java/com/asset/asset_management/service/AssetService.java
package com.asset.asset_management.service;

import com.asset.asset_management.entity.Asset;
import org.springframework.data.domain.Page;

import java.util.List;

public interface AssetService {
    Asset saveAsset(Asset asset);
    List<Asset> getAllAssets();
    Asset getAssetById(Long id);
    Asset updateAsset(Long id, Asset asset);
    void deleteAsset(Long id);
    Page<Asset> getAssetsPaged(int page, int size);
}

File: asset-ui/src/services/AssetService.js
import axios from "axios";

const API_BASE_URL = "http://localhost:8001/api/assets";

class AssetService {

  getAssets(page = 0, size = 5) {
    return axios.get(`${API_BASE_URL}?page=${page}&size=${size}`);
  }

  getAssetById(id) {
    return axios.get(`${API_BASE_URL}/${id}`);
  }

  createAsset(asset) {
    return axios.post(API_BASE_URL, asset);
  }

  updateAsset(id, asset) {
    return axios.put(`${API_BASE_URL}/${id}`, asset);
  }

  deleteAsset(id) {
    return axios.delete(`${API_BASE_URL}/${id}`);
  }
}

export default new AssetService();
